src/antsibull_core/subprocess_util.py

```diff
@@ -87,6 +87,28 @@ async def _stream_log(
     return "".join(lines)
 
 
+def _get_log_func_and_prefix(
+    name: str, loglevel: str | OutputCallbackType | None, logger: Any
+) -> tuple[OutputCallbackType | None, str]:
+    logfunc: Callable[[str], Any] | None = None
+    log_prefix = f"{name}: "
+    if loglevel:
+        if callable(loglevel):
+            logfunc = loglevel
+            log_prefix = ""
+        else:
+
+            def logfunc(string: str, /):
+                func = getattr(logger, loglevel)
+                if isinstance(logger, TwiggyLogger):
+                    return func("{0}", string)
+                if isinstance(logger, StdLogger):
+                    return func("%s", string)
+                return func(string)
```

Comment:
This feels somewhat inefficient. Every time when a log message should be printed, up to two `instanceof()`s are evaluated. I would do the casting and the `getattr` in `_get_log_func_and_prefix` (as before) and create a different function depending on what `logger` is.